src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -277,15 +296,16 @@ private void TextBuffer_OnChanged(object sender, TextContentChangedEventArgs arg
                 StopIdleTimer();
             }
 
+            var snapshot = args.After;
             if (!args.TextChangeOccurred(out var changeInformation))
             {
+                // Ensure we get a parse for latest snapshot.
+                QueueChange(null, snapshot);
```

Comment:
Actually I'm not sure about this. If there was no text change, why do we need to queue a full reparse?

Reply:
There are completion scenarios when other completion providers trigger a text buffer change that has 0 text changes. In those cases the completion async API uses that 0 change snapshot to ask for completions at which point if we don't have a parse flowing through they system we deadlock. The other option we could have taken would be to update the "latest parsed snapshot" to be the one passed into the TextBuffer OnChanged event; however, because of [this](https://github.com/aspnet/AspNetCore/issues/6688) issue we'll have to queue a full reparse anyways so figured I wouldn't write dead code 😄